### DISCUSSION TITLE
Is it possible to conditionally configure modules based on the directory?

### DISCUSSION THREAD

**Discussion by rcedwards** [starship/starship#4363] [Q&A]
I wasn't able to find an answer to this in the configuration docs or by searching issues/discussion here.

I have a large network file system that makes multiple modules run very long ex.:

```
 Here are the timings of modules in your prompt (>=1ms or output):
 git_status  -  1218ms  -   ""
 git_branch  -  1101ms  -   ""
 directory   -  1090ms  -   "redacted/example "
 git_commit  -  1059ms  -   ""
 git_state   -   993ms  -   ""
 package     -    15ms  -   ""
 line_break  -    <1ms  -   "\n"
 character   -    <1ms  -   "$ "
```

It would be great if I could leave these modules enabled for use elsewhere on my system but have them disabled for a specific root path.

I'm wondering if this is already possible or something that could be added in the future?

Update: 

From doing a bit of debugging it would also seem like the directory module should honor both the option `truncate_to_repo = false` as well as disabling the entire `git_` modules when its computing the path here: https://github.com/starship/starship/blob/master/src/modules/directory.rs#L53

This appears to be why the directory command takes a full second in this large directory.

**Comment by a-b-r-o-w-n** [comment]
I'm also looking for a way to do this. Coming from Spaceship prompt, I had env variables set to turn off certain modules for large repositories. A simple suggestion would be to allow for directory level config files so that I can override.

**Comment by jankatins** [comment]
There is https://github.com/starship/starship/pull/4439 which would enable setting config via env vars and therefore would work nicely via direnv and similar solutions which set env vars when you CD into a directory.

If you think that would work for you, please leave your thumbs up in the PR :-)

**Comment by vergenzt** [comment]
I'm also looking to do this! But my use case is at https://github.com/starship/starship/issues/6076#issuecomment-2724814116

TL;DR: I'm adding a custom module for https://github.com/jj-vcs/jj which is a git-compatible VCS / git frontend. Preferably I'd like to disable the `git_*` modules by a custom command (`jj root --ignore-working-copy`, the same `when` command I'm using for my `custom.jj` module) -- but disabling `git_*` modules via environment variables would be better than nothing.

**Reply by johnstegeman** [reply]
One thing I've found with this approach... Custom modules dont work so well with the fill module https://github.com/starship/starship/discussions/7007

**Comment by jonas-lomholdt** [comment]
Yes! I use both Google Cloud and Azure, and I want specific projects to enable the azure module, and others to enable gcloud. Maybe `enabled_on_paths` and then have it disabled in general.

**Reply by bwiggs** [reply]
Yes! I use the AWS CLI for some projects, but i don't want to always see the default AWS profile info in my prompt.
